packages/framework/src/Component/Redis/RedisFacade.php

```diff
@@ -52,10 +52,25 @@ public function cleanCache(): void
         foreach ($this->getCacheClients() as $redis) {
             $prefix = (string)$redis->getOption(Redis::OPT_PREFIX);
             $pattern = $prefix . '*';
-            if (!$this->hasAnyKey($redis, $pattern)) {
-                continue;
+            $this->cleanCacheByScan($redis, $pattern);
+        }
+    }
+
+    /**
+     * @param \Redis $redisClient
+     * @param string $pattern
+     */
+    protected function cleanCacheByScan(Redis $redisClient, string $pattern): void
+    {
+        $suggestedScanBatchSize = 1000;
+
+        $iterator = null;
+        $keys = null;
+        while ($keys !== false) {
+            $keys = $redisClient->scan($iterator, $pattern, $suggestedScanBatchSize);
+            if (is_array($keys) && count($keys) > 0) {
+                $redisClient->eval("return redis.call('unlink', ARGV[1])", $keys);
```

Comment:
why `eval` and not `unlink` method directly? I would like to know if it's for some specific reason

Reply:
Hello, that's because Redis clients are weird. `$redisClient->scan()` will return all keys with some pattern, but when I want to use `$redisClient->unlink()` I need to remove prefix from all keys to be able to unlink them (Prefix is defined in `snc_redis.yml` file). Instead of removing prefix I prefer to use `$redisClient->eval()` to circumvent this behavior.

Reply:
And I had that `eval` function wrong. It was removing only first key. Now it's working as expected

Reply:
Thank you 👍